ncm-metaconfig/src/main/perl/metaconfig.pm

```diff
@@ -123,7 +124,13 @@ sub handle_service
         return;
     }
 
-    $self->prepare_action($srv, $file, $actions) if ($fh->close());
+    my $changes = $fh->close();
+    if ( $changes > 0 ) {
```

Comment:
In fact, I should have! Must test for negative return (error) which is not a success...